pkg/detectors/uri/uri.go

```diff
@@ -19,85 +24,113 @@ type Scanner struct {
 }
 
 // Ensure the Scanner satisfies the interface at compile time.
-var _ detectors.Detector = (*Scanner)(nil)
-var _ detectors.CustomFalsePositiveChecker = (*Scanner)(nil)
+var _ interface {
+	detectors.Detector
+	detectors.CustomFalsePositiveChecker
+} = (*Scanner)(nil)
 
 var (
 	keyPat = regexp.MustCompile(`\b(?:https?:)?\/\/[\S]{3,50}:([\S]{3,50})@[-.%\w\/:]+\b`)
 
 	// TODO: make local addr opt-out
 	defaultClient = detectors.DetectorHttpClientWithNoLocalAddresses
+
+	hostNotFoundCache = simple.NewCache[struct{}]()
```

Comment:
I think this would benefit from being centralized and shared by all detectors. A domain that's found to not exist for [Azure Container Registry](https://github.com/trufflesecurity/trufflehog/blob/ab36b3775e0499be1adc9dd8d3a245236a5671df/pkg/detectors/azurecontainerregistry/azurecontainerregistry.go#L35), [Azure OpenAI](https://github.com/trufflesecurity/trufflehog/blob/ab36b3775e0499be1adc9dd8d3a245236a5671df/pkg/detectors/azure_openai/azure_openai.go#L36), JDBC (below), etc. is applicable to any other detector.

```
Found unverified result 🐷🔑❓
Verification issue: lookup your_server.database.windows.net: no such host
Detector Type: JDBC
Decoder Type: PLAIN
Raw result: jdbc:sqlserver://your_server.database.windows.net:1433;
Commit: aa081336863641da6061a892c7304f9823b4e8d6
Commit_source: refs/remotes/origin/pull/2/head (hidden ref)
Email: Brian Gianforcaro <bgianf@microsoft.com>
File: Java/sample_java.java
Line: 11
Link: https://github.com/Azure/azure-sql-database-samples/blob/aa081336863641da6061a892c7304f9823b4e8d6/Java/sample_java.java#L11
Repository: https://github.com/Azure/azure-sql-database-samples.git
Timestamp: 2015-09-28 17:26:38 +0000
```